src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts

```diff
@@ -1,55 +1,56 @@
-import { commitMutation, graphql, Environment } from "relay-runtime"
-import { useSystemContext } from "v2/System"
+import { commitMutation, graphql } from "relay-runtime"
 import {
   SubmitInquiryRequestMutationInput,
   useArtworkInquiryRequestMutation,
 } from "v2/__generated__/useArtworkInquiryRequestMutation.graphql"
+import { useInquiryContext } from "./useInquiryContext"
 
 type UseArtworkInquiryRequestInput = Omit<
   SubmitInquiryRequestMutationInput,
   "inquireableID" | "inquireableType" | "message"
 > & {
-  relayEnvironment?: Environment
   artworkID: string
   message: string
   contactGallery?: boolean | null
 }
 
 export const useArtworkInquiryRequest = () => {
-  const { relayEnvironment: defaultRelayEnvironment } = useSystemContext()
+  const { relayEnvironment } = useInquiryContext()
```

Comment:
Why do we pull the `relayEnvironment` from inquiry context rather than system context?

Reply:
This gets set in login/sign up. We authenticate the environment and the one in system context is still logged out.

Reply:
Another thought is: can we move this behavior up to the system context itself and add the function there? (This would resolve other questions in PR, as code comments there would be unnecessary.) Thinking through it, currently when we log out we do a hard reload, but in the future that doesn't have to be so, and in the meantime we could at least provide a mechanism to refresh relay's environment from where it's typically accessed.

Reply:
Explored that a bit; it won't work because we'd need to return the environment ref rather than the environment itself. Meaning it'd lead to a much larger refactor. It's a good idea to figure this out but I'm just going to add some comments here for now.

Reply:
Np 👍